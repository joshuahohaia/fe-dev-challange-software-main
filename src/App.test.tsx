import { render, screen } from "@testing-library/react";
import App from "./App";

describe("App Component", () => {
  test("renders without crashing", () => {
    render(<App />);
  });

  test("renders Home component", () => {
    render(<App />);
    const homeElement = screen.getByText(/Estate Agent Property Manager/i);
    expect(homeElement).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
});
