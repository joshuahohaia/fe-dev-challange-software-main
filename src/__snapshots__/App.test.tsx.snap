// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App Component > matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <nav
      aria-label="Main navigation"
      class="bg-white p-4 flex justify-between items-center border-b border-gray-200"
    >
      <div
        aria-level="1"
        class="text-black font-sans text-2xl"
        role="heading"
      >
        Estate Agent Property Manager
      </div>
    </nav>
    <div
      aria-label="Property list"
      class="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 gap-4 p-4"
      role="list"
    />
  </div>
</DocumentFragment>
`;
